src/js/containers/search/filters/AgencyListContainer.jsx

```diff
@@ -0,0 +1,134 @@
+            }
+
+            const agencySearchParams = {
+                fields: fieldType,
+                value: input,
+                mode: "contains"
+            };
+
+            this.agencySearchRequest = SearchHelper.fetchAgencies(agencySearchParams);
+
+            this.agencySearchRequest.promise
+                .then((res) => {
+                    const data = res.data.results;
+                    let autocompleteData = [];
+                    // Filter out any selectedAgencies that may be in the result set
+                    if (this.props.agencyType === "Awarding") {
+                        if (this.props.selectedAwardingAgencies.size > 0) {
+                            autocompleteData =
+                            _.differenceWith(data.awarding_agency__subtier_agency__name,
+                                this.props.selectedAwardingAgencies.toArray(), _.isEqual);
+                            this.props.setAutocompleteAwardingAgencies(autocompleteData);
+                        }
+                        else {
+                            this.props.setAutocompleteAwardingAgencies(
+                                data.awarding_agency__subtier_agency__name);
+                        }
+                    }
+                    else if (this.props.agencyType === "Funding") {
+                        if (this.props.selectedFundingAgencies.size > 0) {
+                            autocompleteData =
+                            _.differenceWith(data.funding_agency__subtier_agency__name,
+                                this.props.selectedFundingAgencies.toArray(), _.isEqual);
+                            this.props.setAutocompleteFundingAgencies(autocompleteData);
+                        }
+                        else {
+                            this.props.setAutocompleteFundingAgencies(
+                                data.funding_agency__subtier_agency__name);
+                        }
+                    }
+                });
+        }
+    }
+
+    handleTextInput(agencyInput) {
+        // Clear existing agencies to ensure user can't select an old or existing one
+
+        // Grab input, clear any exiting timeout
+        const input = agencyInput.target.value;
+        window.clearTimeout(this.timeout);
+
+        // Perform search if user doesn't type again for 300ms
+        this.timeout = window.setTimeout(() => {
+            this.queryAutocompleteAgencies(input);
+        }, 300);
+    }
+
+    render() {
+        return (
+            <AgencyList
+                {...this.props}
+                formatter={this.dataFormatter}
+                handleTextInput={this.handleTextInput}
+                onSelect={this.props.selectAgency}
+                placeHolder={this.props.agencyType}
+                selectedFundingAgencies={this.props.selectedFundingAgencies}
+                selectedAwardingAgencies={this.props.selectedAwardingAgencies}
```

Comment:
Since this container knows what agency type it is responsible for, the child `AgencyList` component should only receive data and props related to its specific type (funding/awarding) rather than everything. Eliminating unused props reduces unnecessary re-renders and improves performance.